File: app/src/main/java/command/ceiba/com/co/dojocommandpattern/FirstActivityCommand.java
package command.ceiba.com.co.dojocommandpattern;

import android.content.Context;
import android.content.Intent;

import java.util.Map;

public class FirstActivityCommand extends BaseActivityCommand implements ICommand {

    public FirstActivityCommand(Context context, Map<String, String> notification){
        super(context, notification);
    }

    @Override
    public void execute() {
        Intent intent = new Intent(context, FirstActivity.class);
        receiver.buildNotification(notification.get("title"), notification.get("body"), intent);
    }
}

File: app/src/main/java/command/ceiba/com/co/dojocommandpattern/FirstActivityCommand.kt
package command.ceiba.com.co.dojocommandpattern

import android.content.Context
import android.content.Intent

class FirstActivityCommand(
    private val context: Context,
    private val notification: Map<String, String>,
    private val receiver: NotificationReceiver
    ) : ICommand {

    override fun execute() {
        val intent = Intent(context, FirstActivity::class.java)
        receiver.buildNotification(notification["title"], notification["body"], intent)
    }
}